easybuild/tools/job/gc3pie.py

```diff
@@ -27,63 +27,79 @@
 Interface for submitting jobs via GC3Pie.
 
 @author: Riccardo Murri (University of Zurich)
+@author: Kenneth Hoste (Ghent University)
 """
-
-
 import os
 import time
 
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import build_option
+from easybuild.tools.job.backend import JobBackend
+
+
+_log = fancylogger.getLogger('gc3pie', fname=False)
+
+
 try:
     import gc3libs
     from gc3libs import Application, Run, create_engine
     from gc3libs.core import Engine
     from gc3libs.quantity import hours as hr
     from gc3libs.workflow import DependentTaskCollection
-    HAVE_GC3PIE = True
-except ImportError:
-    HAVE_GC3PIE = False
-
-from easybuild.tools.build_log import print_msg
-from easybuild.tools.config import build_option
-from easybuild.tools.job.backend import JobBackend
-
-from vsc.utils import fancylogger
 
-if HAVE_GC3PIE:
     # inject EasyBuild logger into GC3Pie
     gc3libs.log = fancylogger.getLogger('gc3pie', fname=False)
     # make handling of log.error compatible with stdlib logging
     gc3libs.log.raiseError = False
 
+    # GC3Pie is available, no need guard against import errors
+    def gc3pie_imported(fn):
+        """No-op decorator."""
+        return fn
+
+except ImportError as err:
+    _log.debug("Failed to import gc3libs from GC3Pie."
+               " Silently ignoring, this is a real issue only when GC3Pie is used as backend for --job")
+
+    # GC3Pie not available, turn method in a raised EasyBuildError
+    def gc3pie_imported(_):
+        """Decorator which raises an EasyBuildError because GC3Pie is not available."""
+        def fail(_):
```

Comment:
You want this to be `fail(*args, **kwargs)` since you are applying it to functions with different signature (e.g., `init(self)` and `make_job(self, script, name, ...)`.